video/out/drm_atomic.c

```diff
@@ -458,3 +485,96 @@ bool drm_mode_destroy_blob(int fd, struct drm_mode *mode)
 
     return (ret == 0);
 }
+
+static inline uint16_t clamp16(float x)
+{
+    return (uint16_t)MPCLAMP(x, 0.0, 65535.0);
+}
+
+static inline uint16_t clamp16_round(float x)
+{
+    return clamp16(roundf(x * 50000.0));
+}
+
+int drm_create_hdr_metadata(struct drm_atomic_context *ctx, struct drm_hdr *hdr_data, bool reset)
+{
+#if HAVE_DRM_HDR
+    if (reset)
+        goto out;
```

Comment:
I'm pretty sure this shouldn't even compile, you're jumping over the definition of `metadata` into code that *uses* `metadata`